nomenclature/testing.py

```diff
@@ -36,11 +36,14 @@ def assert_valid_structure(path: Path):
         :class:`DataStructureDefinition`
       - If a `mappings` folder exists, it must be a valid :class:`RegionProcessor`
     """
-    definition = nomenclature.DataStructureDefinition(path / "definitions")
-    if (path / "mappings").is_dir():
+
+    definition = nomenclature.DataStructureDefinition(path / str(definitions))
+    if (path / str(mappings)).is_dir():
         nomenclature.RegionProcessor.from_directory(
-            path / "mappings"
+            path / str(mappings)
         ).validate_mappings(definition)
+    elif mappings != "mappings":
```

Comment:
Not sure I fully understand the logic of this `elif`. Let's say the `mappings` variable is indeed `"mappings"` but such a folder does not exist.
Then `if (path / str(mappings)).is_dir()` would be false but so would be `elif mappings != "mappings"` which means we would not get the error that we would expect. 
Perhaps, this would be better?
```suggestion
    else: # if we don't find the given directory just throw an error
```